### ISSUE TITLE
Improve logic for posted comments filter

### ISSUE THREAD

**Issue by smacker** [src-d/lookout#455]
Currently, we filter out comments with the same text posted on the same line. It works great most of the time. But not in case of sequential commits which "move" changes from the previous commits.
Good example: https://github.com/src-d/lookout/pull/442

I propose to improve this logic by also checking:
- remember line number "from" and the line content
- check the content of new line
- if they are the same: skip the comment

Example in screenshots:

First comment:
<img width="309" alt="screenshot 2019-01-09 at 16 19 29" src="https://user-images.githubusercontent.com/406916/50908704-8be91100-142a-11e9-9693-153cac1a2193.png">

We can skip this one if apply logic from above:
<img width="334" alt="screenshot 2019-01-09 at 16 19 36" src="https://user-images.githubusercontent.com/406916/50908709-8e4b6b00-142a-11e9-8193-c8763a030aea.png">

**Comment by dpordomingo** [comment]
I like the proposal; @carlosms should we consider it, or do you have any other idea?

**Comment by se7entyse7en** [comment]
I agree with the fact that the same comment posted twice on different pushes is annoying, but only if it was resolved on the first push. If the issues was not resolved and in the next push that issue was not addressed, then there's the risk that it was simply ignored or forgotten.

I'm also thinking of another scenario. Let's suppose that the first push contains this code:
```
func FooInc() {
	fn := func(x int) int {  // <--- Lookout-bot: "This line has something I don't like"
		return x + 1
	}
	Bar(fn)
}

func Bar(fn func(int) int) {
	fmt.Println(fn(0))
}
```
with the second line commented by the bot. Now let's say that another push occurs and now the code contains:
```
func FooInc() {
	fn := func(x int) int {  // <--- 
		return x + 1
	}
	Bar(fn)
}

func FooDec() {
	fn := func(x int) int {  // <---
		return x - 1
	}
	Bar(fn)
}

func Bar(fn func(int) int) {
	fmt.Println(fn(0))
}
```
Given that the two marked lines are identical what should we do in this case? Should we skip the comment for both lines? Should we comment both liens? In my opinion we should skip in the first case and notify in the second case. But still only if in the first case it was resolved as I previously said.

I know that this is an artificial example and it could actually never happen and that the strategy of remembering the line number is already an improvement, and maybe there are also other not too complex improvements that we could apply. I'm just want to say that we should consider a trade-off between false positives and false negatives.

**Comment by se7entyse7en** [comment]
Ok, yes I understood that (but thanks for pointing where it's done 👍). But I was referring to what should we do in case that in the secondo push both lines are in different position than the first push.
